### DISCUSSION TITLE
How to let a Quad look always at the camera?

### DISCUSSION THREAD

**Discussion by payload** [bevyengine/bevy#2979] [Q&A] [answered]
I want to achieve the effect where a flat shape in 3d is always aligned to the camera such that it stays a flat shape, like in old games a sprite in 3d.
For that I use a Quad mesh. I think its front face is looking into positive Z direction.
I thought about somehow taking the cameras GlobalTransform rotation after `TransformSystem::TransformPropagate` ran
which "looks" somewhere and somehow invert the direction of looking, while maintaining the up direction.

**Comment by Nilirad** [comment, marked as answer]
What you're referring to is called billboarding. It is usually achieved using a custom vertex shader. For example:
http://www.opengl-tutorial.org/intermediate-tutorials/billboards-particles/billboards/

If you want to rotate the quad instead, you can do that, it's just a bit unusual, but it will still work unless you have multiple cameras looking at the same sprite.
If I remember correctly, the rotation field of the `Transform` component has a `look_at` method (or constructor, in that case you have to do `Quat::look_at`), and put as the parameter the position of the camera.

**Reply by payload** [reply]
Thanks, I can now recall the name billboarding again. Head it before but it is long ago. The article is good and I found that https://github.com/lassade/bevy_gizmos contains a billboarding feature too.
